react/src/components/atoms/table/Table/index.js

```diff
@@ -11,68 +12,84 @@ const Table = (table) => {
-        </thead>
-      }
-      { table.bodies.map((body, bodyIndex) => (
-        <tbody key={body.key || `tbody${bodyIndex}`}>
-          { body.rows.map((row, rowIndex) => {
-            const tableIndexOffset = row.rowSpanOffset ? 0 : 1;
-            const defaultRowKey = `tbody${bodyIndex}-row${rowIndex}`;
-            return(
-              <tr key={row.key || defaultRowKey} className={row.rowSpanOffset ? 'is-offset' : null}>
-                { row.cells.map((cell, cellIndex) => {
-                  let dataLabel = null;
-                  if (tableHeaderRow.length > 0) {
-                    // Mayflower twig loops started at 1 - add one to index.
-                    dataLabel = tableHeaderRow.cells[(cellIndex + 1) - tableIndexOffset].text;
-                  }
-                  const defaultCellKey = `${defaultRowKey}-cell${cellIndex}`;
-                  return(cell.heading ? (
+    <div className="ma__table--responsive js-ma-responsive-table">
+      <div className={responsiveClasses} role="region" id={table.id}>
+        <table className={`ma__table${tableWideClass}`}>
+          {
+            (table.head && table.head.rows) &&
+            <thead>
+              { table.head.rows.map((row, rowIndex) => (
+                <tr key={row.key || `thead-row${rowIndex}`}>
+                  { row.cells.map((cell, cellIndex) => (
                     <th
-                      key={cell.key || defaultCellKey}
-                      data-label={dataLabel}
-                      scope="row"
+                      key={cell.key || `thead-row${rowIndex}-cell${cellIndex}`}
+                      scope="col"
                       colSpan={cell.colspan || null}
                       rowSpan={cell.rowspan || null}
                     >{cell.text}
                     </th>
-                  ) : (
-                    <td
-                      key={cell.key || defaultCellKey}
-                      data-label={dataLabel}
-                      colSpan={cell.colspan || null}
-                      rowSpan={cell.rowspan || null}
-                    >
-                      {cell.text}
-                    </td>
-                  ));
-              })}
-              </tr>);
-          })}
-        </tbody>
-        ))}
-    </table>
+                ))}
+                </tr>
+              ))}
+            </thead>
+        }{table.bodies.map((body, bodyIndex) => (
+          <tbody key={body.key || `tbody${bodyIndex}`}>
+            { body.rows.map((row, rowIndex) => {
+              const tableIndexOffset = row.rowSpanOffset ? 0 : 1;
+              const defaultRowKey = `tbody${bodyIndex}-row${rowIndex}`;
+              return(
+                <tr key={row.key || defaultRowKey} className={row.rowSpanOffset ? 'is-offset' : null}>
+                  { row.cells.map((cell, cellIndex) => {
+                    let dataLabel = null;
+                    if ((cellIndex > 0) && tableHeaderRow.cells && tableHeaderRow.cells.length > 0) {
```

Comment:
tableHeaderRow is not defined?

Reply:
It's defined on line 10

Reply:
i'm seeing this error when i change rowSpanOffset to true on line 30 in Table.knobs.options.js
<img width="894" alt="screen shot 2019-02-06 at 11 23 42 am" src="https://user-images.githubusercontent.com/5789411/52356230-db0e6b80-2a01-11e9-8709-a294aa323eeb.png">